src/aws/flb_aws_util.c

```diff
@@ -927,3 +932,86 @@ flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag,
         }
         return NULL;
 }
+
+/*
+ * Replace %f and %L in time_key_format with milliseconds and nanoseconds. And get the
+ * time format output and output length with strftime().
+ */
+size_t create_time_formatter(char **out_buf, const char *time_key_format,
```

Comment:
This comment isn't exactly the contract of your function. It's an extension to strftime that also supports %f and %L, and the outputted length is that of the formatted time string. Please update this to be exact and in your own words.

Reply:
nit: Also, as noted before, this function has not concept of keys, and it could be re-used in other contexts, so I think the parameter should just be `time_format`